errbot/backends/slack.py

```diff
@@ -650,16 +651,16 @@ def build_identifier(self, txtrep):
         log.debug("building an identifier from %s" % txtrep)
         username, userid, channelname, channelid = self.extract_identifiers_from_string(txtrep)
 
+        if userid is None and username is not None:
+            userid = self.username_to_userid(username)
+        if channelid is None and channelname is not None:
+            channelid = self.channelname_to_channelid(channelname)
+        if userid is not None and channelid is not None:
+            return SlackRoomOccupant(self.sc, userid, channelid, bot=self)
```

Comment:
But that can't be represented in the input string of `build_identifier`, can it? The list of formats this function must support is defined in the documentation (and the tests) as:

- `<#C12345>` (Slack Person with no `userid` but a `channelid`)
- `<@U12345>` (Slack Person with a `userid` and a `channelid` set to its one to one channel)
- `@user` (Slack Person with a `userid` as retrieved from the user name, and a `channelid` set to its one to one channel)
- `#channel` (Slack Person with no `userid` and a `channelid` retrieved from the channel name)
- `#channel/user` (Slack Room Occupant with a `userid` retrieved from the user name and a `channelid` retrieved from the channel name).

The original code can never return a Slack Room Occupant, because in the last format, both a channel name and a user name is set, causing the method to ignore the channel name and return a Slack Person for the user. That's the bug I'm trying to fix.